docs/operating/upgrading.md

```diff
@@ -0,0 +1,110 @@
+---
+sidebar_position: 5
+---
+
+# Upgrading
+
+When upgrading TigerBeetle, each release specifies two important versions:
+* the oldest release that can be upgraded from and,
+* the oldest supported client version.
+
+The upgrade process involves first upgrading the replicas, followed by upgrading the clients. The
+client version *cannot* be newer than the replica version, and will fail with an error message if
+so. Provided the supported version ranges overlap, coordinating the upgrade between clients and
+replicas is not required.
+
+While not strictly necessary to upgrade clients if they are newer than the cluster's oldest
+supported client version, it is recommended both to access new features, and to keep the upgrade
+path simple in future. If the client version falls behind too far, a multistep upgrade process is
+required.
+
+## Planning for upgrades
+Upgrading causes a short period of unavailability as the replicas restart. This is on the order of
+5 seconds, and will show up as a latency spike on requests. The TigerBeetle clients will internally
+retry any requests during the period.
+
+Upgrading from 0.15.3 to 0.15.4 requires a slightly longer period, as it's necessary to manually
+restart all replicas.
+
+Even though this period is short, scheduling a maintenance windows for upgrades is still
+recommended, for an extra layer of safety.
+
+## Upgrading binary-based installations
+If TigerBeetle is installed under `/usr/bin/tigerbeetle`, and you wish to upgrade to `0.15.4`:
+```bash
+# SSH to each replica, in no particular order:
+cd /tmp wget
+https://github.com/tigerbeetle/tigerbeetle/releases/download/0.15.4/tigerbeetle-x86_64-linux.zip
+unzip tigerbeetle-x86_64-linux.zip
+
+# Put the binary on the same file system as the target, so mv is atomic.
+mv tigerbeetle /usr/bin/tigerbeetle-new
+
+rm /usr/bin/tigerbeetle mv /usr/bin/tigerbeetle-new /usr/bin/tigerbeetle
```

Comment:
same here

Reply:
I'd maybe non-desttructively `mv` the old beetle, in case anything goes wrong and you would like to know what you've been running before the upgrade.